plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/BaseRedshiftCastPushdownTest.java

```diff
@@ -0,0 +1,415 @@
+                new CastTestCase("c_varchar_10", "varchar(50)", Optional.of("c_nvarchar_50")),
+                new CastTestCase("c_varchar_10", "varchar(65535)", Optional.of("c_nvarchar_65535")),
+                new CastTestCase("c_varchar_10", "varchar(256)", Optional.of("c_text")),
+
+                new CastTestCase("c_varchar_unicode", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_nvarchar_unicode", "varchar(50)", Optional.of("c_varchar_50")),
+
+                new CastTestCase("c_boolean", "bigint", Optional.of("c_bigint")),
+                new CastTestCase("c_smallint", "bigint", Optional.of("c_bigint")),
+                new CastTestCase("c_integer", "bigint", Optional.of("c_bigint")),
+                new CastTestCase("c_bigint", "bigint", Optional.of("c_bigint")),
+                new CastTestCase("c_decimal_10_2", "bigint", Optional.of("c_bigint")),
+                new CastTestCase("c_decimal_19_2", "bigint", Optional.of("c_bigint")),
+                new CastTestCase("c_decimal_30_2", "bigint", Optional.of("c_bigint")),
+                new CastTestCase("c_decimal_negative", "bigint", Optional.of("c_bigint")));
+    }
+
+    @Override
+    protected List<CastTestCase> unsupportedCastTypePushdown()
+    {
+        return ImmutableList.of(
+                new CastTestCase("c_real", "smallint", Optional.of("c_smallint")),
+                new CastTestCase("c_float4", "smallint", Optional.of("c_smallint")),
+                new CastTestCase("c_double_precision", "smallint", Optional.of("c_smallint")),
+                new CastTestCase("c_float", "smallint", Optional.of("c_smallint")),
+                new CastTestCase("c_float8", "smallint", Optional.of("c_smallint")),
+                new CastTestCase("c_varchar_numeric", "smallint", Optional.of("c_smallint")),
+                new CastTestCase("c_text_numeric", "smallint", Optional.of("c_smallint")),
+                new CastTestCase("c_nvarchar_numeric", "smallint", Optional.of("c_smallint")),
+                new CastTestCase("c_varchar_numeric_sign", "smallint", Optional.of("c_smallint")),
+
+                new CastTestCase("c_smallint", "boolean", Optional.of("c_boolean")),
+                new CastTestCase("c_real", "double", Optional.of("c_double_precision")),
+                new CastTestCase("c_double_precision", "real", Optional.of("c_real")),
+                new CastTestCase("c_double_precision", "decimal(10,2)", Optional.of("c_decimal_10_2")),
+
+                new CastTestCase("c_varchar_unicode", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_nvarchar_unicode", "char(50)", Optional.of("c_char_50")),
+
+                new CastTestCase("c_timestamp", "date", Optional.of("c_date")),
+                new CastTestCase("c_timestamp", "time", Optional.of("c_time")),
+                new CastTestCase("c_date", "timestamp", Optional.of("c_timestamp")));
```

Comment:
Some unsupported CASTs to varchar will land here, too, e.g.:
CAST(c_decimal_10_2 AS varchar(50))